### PR TITLE
rate counter fix race

### PR DESCRIPTION
**Pull request by tjannaud**
.counter is not used behind the lock so we need the atomic lib

### CODE REVIEW

File: rate_counter.go [hunk 1]
```diff
@@ -84,133 +84,131 @@ func NewStandardRateCounter(numSamples int64, samplePeriodMs int64, clock clock.
...
-	defer this.lock.RUnlock()
+func (sRC *StandardRateCounter) Snapshot() RateCounter {
+	sRC.maybeSampleCount()
+	sRC.lock.RLock()
+	defer sRC.lock.RUnlock()
 
 	return &RateCounterSnapshot{
-		count: this.counter,
-		rate:  this.lastRate,
+		count: atomic.LoadInt64(&sRC.counter),
```
Comment (tjannaud) [inline rate_counter.go]: here

File: rate_counter.go [hunk 2]
```diff
@@ -84,133 +84,131 @@ func NewStandardRateCounter(numSamples int64, samplePeriodMs int64, clock clock.
...
+	defer sRC.lock.RUnlock()
+	return sRC.lastRate
 }
 
-func (this *StandardRateCounter) Count() int64 {
-	this.lock.RLock()
-	defer this.lock.RUnlock()
-	return this.counter
+func (sRC *StandardRateCounter) Count() int64 {
+	return atomic.LoadInt64(&sRC.counter)
```
Comment (tjannaud) [inline rate_counter.go]: here

File: rate_counter.go [hunk 3]
```diff
@@ -84,133 +84,131 @@ func NewStandardRateCounter(numSamples int64, samplePeriodMs int64, clock clock.
 	return rc
 }
 
-func (this *StandardRateCounter) Clear() {
-	this.lock.Lock()
-	defer this.lock.Unlock()
+func (sRC *StandardRateCounter) Clear() {
```
Comment (logan) [inline rate_counter.go]: nit: don't capitalize initialisms (`src`)

Would be better to do this style change as its own PR.

**Review by logan — APPROVED** [review]
